### DISCUSSION TITLE
Running marqo offline

### DISCUSSION THREAD

**Discussion by msabty** [marqo-ai/marqo#1020] [Q&A] [answered]
I am running on an air gapped RHEL node, and I would like to try out marqo. I got pulled the original image from docker hub, and I saw it tried to download the embedding model from huggingface, so I made a new image from the marqo image that copies the model directly to /app/marqo/cache.

 However, I still fail to run it because it tries to get the model information from hugginface again even though the model is cached.

Are there any flags I should be altering to use marqo in offline mode?

**Comment by wanliAlex** [comment]
@msabty Our team is looking at this and will come back to you.

**Reply by msabty** [reply, marked as answer]
Thanks. 

I copied the /app/marqo/cache directory and the ~/.cache directory and set the huggingface offline flag to 1, and now it seems to run without crash it when I run the container with no network but I need to try it in actual usage.

**Reply by wanliAlex** [reply]
Great! Another potential issue is the nltk data that is used to preprocess text. It is under the directory `/root/nltk_data`. You need to copy this directory into your new container. We will fix this in the next release 2.14 and the nltk data will be included in the base image.
